services/searchengine/bleveengine/search.go

```diff
@@ -209,6 +212,72 @@ func (b *BleveEngine) SearchPosts(channels *model.ChannelList, searchParams []*m
 	return postIds, matches, nil
 }
 
+func (b *BleveEngine) DeletePosts(searchRequest *bleve.SearchRequest, batchSize int) (int64, error) {
+	resultsCount := int64(0)
+
+	for {
+		// From is not a offset but from where to start picking results, so
+		// keeping it always to 0 makes the work
```

Comment:
I find the wording slightly confusing as `From` is technically an offset, is just that we're not using it. A different proposal:

```suggestion
		// As we are deleting the posts after fetching them, we need to keep
		// From fixed always to 0
```